### PR TITLE
refactor(AWS API Gateway): Deprecate default `identitySource`

### PR DESCRIPTION
**Pull request by pgrzesik**
Addresses: #9458 

Addresses first part of linked issue by introducing deprecation. The breaking change should be introduced with next major release.

### CODE REVIEW

File: lib/plugins/aws/package/compile/events/apiGateway/index.js [hunk 1]
```diff
@@ -264,6 +264,35 @@ class AwsCompileApigEvents {
...
+                // Authorizer is defined as a string and we want to check if its `aws_iam` - in such case the deprecation will not apply
+                return !(http.authorizer.toUpperCase() === 'AWS_IAM');
+              }
+              return false;
+            })
+          )
+        ) {
+          this.serverless._logDeprecation(
+            'AWS_API_GATEWAY_DEFAULT_IDENTITY_SOURCE',
+            'Starting with `v3.0.0`, `identitySource` property for `http.authorizer` will no longer be set to "method.request.header.Authorization" by default. If you want to keep this setting, please set it explicitly in your configuration.'
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/apiGateway/index.js:292]: I think there are two issues we deal with:

1. We attempt to set default value to `identitySource` (design bug)
2. We treat `undefined` and `null` as set for `identitySource` differently (logic bug)

Probably best way to fix it, is to leave how it works now, and fix both of the issues with next major.

Still the way the deprecation afaik should be communicated is:

```
Starting with v3 not setting "functions[].http.authorizer.identitySource" will no longer result with "method.request.header.Authorization" being as as default.
If you want to keep this setting, please set it explicitly in your configuration. If you don't want this to be set, set it to "null"
```

**Review by medikoo — CHANGES_REQUESTED** [review]

File: lib/plugins/aws/package/compile/events/apiGateway/index.js [hunk 2]
```diff
@@ -264,6 +264,35 @@ class AwsCompileApigEvents {
...
+            events.some(({ http }) => {
+              if (http && http.authorizer) {
+                if (_.isObject(http.authorizer)) {
+                  // Deprecation should not apply to already existing, external authorizers
+                  if (http.authorizer.authorizerId) return false;
+                  // Deprecation should not apply to `aws_iam` authorization type
+                  if (http.authorizer.type && http.authorizer.type.toUpperCase() === 'AWS_IAM') {
+                    return false;
+                  }
+                  if (http.authorizer.identitySource) return false;
```
[the remark this answers is not in the document]
Comment (pgrzesik) [reply lib/plugins/aws/package/compile/events/apiGateway/index.js:279]: This is a great point, I've missed the fact that you can actually turn it off by setting it to `null`

File: lib/plugins/aws/package/compile/events/apiGateway/index.js [hunk 3]
```diff
@@ -264,6 +264,33 @@ class AwsCompileApigEvents {
...
+                // Authorizer is defined as a string and we want to check if its `aws_iam` - in such case the deprecation will not apply
+                return !(http.authorizer.toUpperCase() === 'AWS_IAM');
+              }
+              return false;
+            })
+          )
+        ) {
+          this.serverless._logDeprecation(
+            'AWS_API_GATEWAY_DEFAULT_IDENTITY_SOURCE',
+            'Starting with `v3.0.0`, "functions[].events[].http.authorizer.identitySource" will no longer be set to "method.request.header.Authorization" by default. If you want to keep this setting, please set it explicitly in your configuration. If you do not want this to be set, please set it explicitly to `null`.'
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/apiGateway/index.js:290]: In CLI logs we tend to not use back-ticks, we eventually wrap with regular quotes. Here I think there's no need to quote `v3.0.0`, and I would wrap `null` with quotes.

Let's also maybe add new line after first sentence.

**Review by medikoo — CHANGES_REQUESTED** [review]
@pgrzesik Look great, I've just posted some suggestions towards message formatting

**Review by medikoo — APPROVED** [review]
Looks great 👍

**Pull request merged**